org.eclipse.lemminx/src/main/resources/org/eclipse/lemminx/services/snippets/processing-instruction-snippets.json

```diff
@@ -0,0 +1,63 @@
+{
+	"Insert XML Model Schema Association": {
+	  "prefix": [
+		"<?xml-model"
+	  ],
+	  "suffix": "?>",
+	  "body": [
+		"<?xml-model href=\"${1:location}\"?>${0}"
+	  ],
+	  "label": "$description",
+	  "description": "Insert XML Model Processing Instruction for Schema Association"
+	},
+	"Insert XML Model Schema Association with Type": {
+	  "prefix": [
+		"<?xml-model"
+	  ],
+	  "suffix": "?>",
+	  "body": [
+		"<?xml-model href=\"${1:location}\" type=\"${2|application/xml,application/xml-dtd,application/relax-ng-compact-syntax|}\"?>${0}"
+	  ],
+	  "label": "$description",
+	  "description": "Insert XML Model Processing Instruction for Schema Association including Type"
+	},
+	"Insert XML Model Schema Association with Schematypens": {
+	  "prefix": [
+		"<?xml-model"
+	  ],
+	  "suffix": "?>",
+	  "body": [
+		"<?xml-model href=\"${1:location}\" schematypens=\"${2|http://www.w3.org/2001/XMLSchema,http://relaxng.org/ns/structure/1.0,http://purl.oclc.org/dsdl/schematron,http://purl.oclc.org/dsdl/nvdl/ns/structure/1.0|}\"?>${0}"
+	  ],
+	  "label": "$description",
+	  "description": "Insert XML Model Processing Instruction for Schema Association including Schematypens"
+	},
+	"Insert XML Model Schema Association with Type and Schematypens": {
+	  "prefix": [
+		"<?xml-model"
+	  ],
+	  "suffix": "?>",
+	  "body": [
+		"<?xml-model href=\"${1:location}\" ${2|type=\"application/xml-dtd\",type=\"application/xml\" schematypens=\"http://www.w3.org/2001/XMLSchema\",type=\"application/xml\" schematypens=\"http://relaxng.org/ns/structure/1.0\",type=\"application/relax-ng-compact-syntax\",type=\"application/xml\" schematypens=\"http://purl.oclc.org/dsdl/schematron\",type=\"application/xml\" schematypens=\"http://purl.oclc.org/dsdl/nvdl/ns/structure/1.0\"|}?>${0}"
```

Comment:
Why you have several type= attributes?

Reply:
see my reply above :)

Reply:
> 
> 
> Why you have several type= and schematypens= attributes?

My reasoning was the following: if sou use an xsd schema, you'll allways want `type="application/xml"` and the according schematypens, if you have a dtd, you'll allways want `type="application/xml-dtd"` but you don't want the schematypens at all.  
So instead of having two different choices, where people can choose combinations that don't fit; and to avoid having the schematypens in cases where it can't be (DTD and compact RelaxNG), I just put both attributes in one choice.

I thoght that's actually more reliable and user friendly. But if you don't think it's a good idea, I can easily get rid of it again.